201-key-vault-secret-create/azuredeploy.json

```diff
@@ -2,107 +2,133 @@
       "tags": {
         "displayName": "KeyVault"
       },
       "properties": {
-        "enabledForDeployment": "[parameters('enabledForDeployment')]",
-        "enabledForTemplateDeployment": "[parameters('enabledForTemplateDeployment')]",
-        "enabledForVolumeEncryption": "[parameters('enableVaultForVolumeEncryption')]",
+        "enabledForDeployment": "[variables('enabledForDeployment')]",
+        "enabledForTemplateDeployment": "[variables('enabledForTemplateDeployment')]",
+        "enabledForDiskEncryption": "[variables('enabledForDiskEncryption')]",
         "tenantId": "[parameters('tenantId')]",
-        "accessPolicies": "[parameters('accessPolicies')]",
+        "accessPolicies": [
+          {
+            "objectId": "[parameters('objectId')]",
+            "tenantId": "[parameters('tenantId')]",
+            "permissions": {
+              "keys": [
+                "Get",
+                "List",
+                "Update",
+                "Create",
+                "Import",
+                "Delete",
+                "Recover",
+                "Backup",
+                "Restore"
+              ],
+              "secrets": [
+                "Get",
+                "List",
+                "Set",
+                "Delete",
+                "Recover",
+                "Backup",
+                "Restore"
+              ],
+              "certificates": [
+                "Get",
+                "List",
+                "Update",
+                "Create",
+                "Import",
+                "Delete",
+                "Recover",
+                "Backup",
+                "Restore",
+                "ManageContacts",
+                "ManageIssuers",
+                "GetIssuers",
+                "ListIssuers",
+                "SetIssuers",
+                "DeleteIssuers"
+              ]
+            }
+          }
+        ],
         "sku": {
-          "name": "[parameters('vaultSku')]",
+          "name": "[variables('sku')]",
           "family": "A"
+        },
+        "networkAcls": {
+          "value": {
+            "defaultAction": "Allow",
+            "bypass": "AzureServices",
+            "virtualNetworkRules": [],
+            "ipRules": []
+          }
         }
       }
     },
     {
       "type": "Microsoft.KeyVault/vaults/secrets",
-      "name": "[concat(parameters('keyVaultName'), '/', parameters('secretsObject').secrets[copyIndex()].secretName)]",
-      "apiVersion": "2015-06-01",
+      "name": "[concat(parameters('name'), '/', parameters('secretsObject').secrets[copyIndex()].secretName)]",
+      "apiVersion": "2018-02-14",
+      "dependsOn": [
+        "[concat('Microsoft.KeyVault/vaults/', parameters('name'))]"
+      ],
       "properties": {
         "value": "[parameters('secretsObject').secrets[copyIndex()].secretValue]"
       },
-      "dependsOn": [
-        "[concat('Microsoft.KeyVault/vaults/', parameters('keyVaultName'))]"
-      ],
       "copy": {
```

Comment:
this loop needs an input property

Reply:
Not sure I understand.  The input property is an object of a list secrets.

Reply:
nm, I see it... move "copy" above "properties"...